test/controllers/purchase_controller_test.rb

```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PurchaseControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```

Comment:
- [ ] Let's add unit tests and integration tests for all the most important components of your application.